src/main/java/com/mycompany/shop/basket/BasketController.java

```diff
@@ -62,14 +122,22 @@ public void setId(Long id) {
             itemRepository.save(item);
             basket.getItems().add(item);
             basket.recalculateTotalPrice();
-            basket = basketRepository.save(basket);
             sessionBean.setBasket(basket);
             return ResponseEntity.ok(basket);
         } else {
             throw new ProductNotFoundException("The product with id: " + createBasketRequest.getId() + " is not found");
         }
     }
 
+    @DeleteMapping("/logout")
+    public void clearBasketFromSession() {
+        Basket basket = sessionBean.getBasket();
+        if (basket != null) {
+            sessionBean.setBasket(null);
+        }
+    }
```

Comment:
why here? it should be part of general logout method

Reply:
+